inference-gateway/src/events_reporter.rs

```diff
@@ -0,0 +1,158 @@
+use anyhow::{Context, Result};
+use chrono::{DateTime, Utc};
+use log::info;
+use pgmq::PGMQueueExt;
+use serde::{Deserialize, Serialize};
+use sqlx::{Pool, Postgres};
+use std::{env, time::Duration};
+use tokio::time::interval;
+use uuid::Uuid;
+
+use crate::db::connect;
+use crate::errors::DatabaseError;
+
+pub fn split_events(events: Vec<Events>, max_size: usize) -> Vec<Message> {
+    events
+        .chunks(max_size)
+        .map(|chunk| Message {
+            id: Uuid::new_v4().to_string(),
+            message: chunk.to_vec(),
+        })
+        .collect()
+}
+
+pub async fn get_usage(
+    dbclient: &Pool<Postgres>,
+    start_time: DateTime<Utc>,
+    end_time: DateTime<Utc>,
+) -> Result<Vec<Events>, DatabaseError> {
+    let rows = sqlx::query_as!(
+        UsageData,
+        r#"
+        SELECT
+        organization_id,
+        instance_id,
+        model,
+        MAX(completed_at) AS completed_at,
```

Comment:
I could switch to COALESCE here to avoid `SUM` and `MAX` returning optional fields, not sure which option is more readable though